File: apiwdbe/WdbeQCvrAIp.cpp
/**
	* \file WdbeQCvrAIp.cpp
	* API code for table TblWdbeQCvrAIp (implementation)
	* \author Alexander Wirthmueller
	* \date created: 23 Aug 2020
	* \date modified: 23 Aug 2020
	*/

#include "WdbeQCvrAIp.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQCvrAIp
 ******************************************************************************/

WdbeQCvrAIp::WdbeQCvrAIp(
			const uint jnum
			, const string sref
			, const string srefIxVTagtype
			, const string titIxVTagtype
			, const string Comment
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->srefIxVTagtype = srefIxVTagtype;
	this->titIxVTagtype = titIxVTagtype;
	this->Comment = Comment;
};

bool WdbeQCvrAIp::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQCvrAIp");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "srefIxVTagtype", "tty", srefIxVTagtype);
		extractStringUclc(docctx, basexpath, "titIxVTagtype", "tty2", titIxVTagtype);
		extractStringUclc(docctx, basexpath, "Comment", "cmt", Comment);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQCvrAIp
 ******************************************************************************/

ListWdbeQCvrAIp::ListWdbeQCvrAIp() {
};

ListWdbeQCvrAIp::ListWdbeQCvrAIp(
			const ListWdbeQCvrAIp& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQCvrAIp(*(src.nodes[i]));
};

ListWdbeQCvrAIp& ListWdbeQCvrAIp::operator=(
			const ListWdbeQCvrAIp& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQCvrAIp(*(src.nodes[i]));

	return *this;
};

ListWdbeQCvrAIp::~ListWdbeQCvrAIp() {
	clear();
};

void ListWdbeQCvrAIp::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQCvrAIp::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQCvrAIp* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQCvrAIp");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQCvrAIp", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQCvrAIp(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQCvrAIp[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwdbe/WdbeQCvrAIp.java
/**
	* \file WdbeQCvrAIp.h
	* Java API code for record of table TblWdbeQCvrAIp
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQCvrAIp {
	
	public WdbeQCvrAIp(
				int jnum
				, String sref
				, String srefIxVTagtype
				, String titIxVTagtype
				, String Comment
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.srefIxVTagtype = srefIxVTagtype;
		this.titIxVTagtype = titIxVTagtype;
		this.Comment = Comment;
	};

	public int jnum;
	public String sref;
	public String srefIxVTagtype;
	public String titIxVTagtype;
	public String Comment;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQCvrAIp");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			srefIxVTagtype = Xmlio.extractStringUclc(doc, basexpath, "srefIxVTagtype", "tty", null, 0);
			titIxVTagtype = Xmlio.extractStringUclc(doc, basexpath, "titIxVTagtype", "tty2", null, 0);
			Comment = Xmlio.extractStringUclc(doc, basexpath, "Comment", "cmt", null, 0);
			
			return true;
		};

		return false;
	};
};
